### PR TITLE
Add arithmetic to length units

### PR DESCRIPTION
**Pull request by ch3pjw**
I wanted to be able to add some lengths together without understanding the underlying canonical "user" unit. For example, if I want to specify something as being 3 mm left of a particular x coordinate (say 88 mm), I can write `mm(88) - mm(4)` and get `mm(84)`. We cast to "unitless" user units in the case where you mix unit types. I implemented scalar multiplication and division in a way I hope is sensible. Please pick holes.

### CODE REVIEW

File: svgast/units.py [hunk 1]
```diff
@@ -22,6 +22,66 @@ def __str__(self):
...
+        if type(self) is type(other):
+            return type(self)(l)
+        else:
+            return l
+
+    def __rsub__(self, other):
+        assert not isinstance(other, Length)
+        return user(other - self.px)
+
+    def __mul__(self, other):
```
Comment (foolswood) [inline svgast/units.py:58]: Multiplication and division are interesting for this:
I think `3 * 2cm` makes sense, but I'm not sure that `3cm * 2mm` does make sense.
I presume you're using complex numbers in the usual way to represent a 2d vector?

File: svgast/units.py [hunk 2]
```diff
@@ -22,6 +22,66 @@ def __str__(self):
...
+            raise TypeError()
+
+    def __rmul__(self, other):
+        if isinstance(other, Complex):
+            return type(self)(user(other * self.px))
+        else:
+            raise TypeError()
+
+    def __truediv__(self, other):
+        if isinstance(other, Length):
```
Comment (foolswood) [inline svgast/units.py:71]: This isn't quite the opposite of multiplication?

File: svgast/units.py (hunk 1 above)
Comment (ch3pjw) [reply svgast/units.py:58]: I've explicitly said that `3cm * 2mm` isn't allowed - only multiplication by a scalar. I read somewhere that `Complex` was the way to say "number" in Python - might be wrong on that front! Happy for suggestions.

File: svgast/units.py (hunk 2 above)
Comment (ch3pjw) [reply svgast/units.py:71]: Yeah, it doesn't allow you to flip the operands like multiplication does. Dividing a scalar by a united length seemed weird. I was running off the idea that units like `mm^-1` and `cm^2` were to be avoided, but anything that gave you a regular "1D" unit was fine.

File: svgast/units.py (hunk 1 above)
Comment (foolswood) [reply svgast/units.py:58]: Erm, there's a numbers.Number whose docstring claims "All numbers inherit from this class.". Complex has a conjugate and stuff and does in fact appear to be a complex number.

File: svgast/units.py (hunk 2 above)
Comment (foolswood) [reply svgast/units.py:71]: I just meant it looked different in type terms, although I think duck typing might make it equivalent.

**Review by foolswood — APPROVED** [review]

**Pull request merged**